src/Writer/Renderer/AbstractRenderer.php

```diff
@@ -188,8 +190,10 @@ public function getType()
      * helps simplify the appending of namespace declarations, but also ensures
      * namespaces are added to the root element - not scattered across the entire
      * XML file - may assist namespace unsafe parsers and looks pretty ;).
+     *
+     * @return void
      */
-    public function setRootElement(DOMElement $root)
+    public function setRootElement(DOMElement $root): void
```

Comment:
Please do not add php typehints as this will is a BC break unless the class is final.